core/src/execution/datafusion/planner.rs

```diff
@@ -1094,8 +1094,15 @@ impl PhysicalPlanner {
     ) -> Result<Arc<dyn AggregateExpr>, ExecutionError> {
         match spark_expr.expr_struct.as_ref().unwrap() {
             AggExprStruct::Count(expr) => {
-                let child = self.create_expr(&expr.children[0], schema)?;
-                Ok(Arc::new(Count::new(child, "count", DataType::Int64)))
+                let mut children = vec![];
+                for child in expr.children.iter() {
+                    children.push(self.create_expr(child, schema.clone())?);
+                }
```

Comment:
This could be written in a functional style to avoid mutating state.

```suggestion
                let children = expr
                    .children
                    .iter()
                    .map(|child| self.create_expr(child, schema.clone()))
                    .collect::<Result<Vec<_>, _>>()?;
```